File: A_New_Year_Candles.cpp
#include <iostream>
using namespace std;

int main()
{
    int a, b;
    cin >> a >> b;

    int ans = 0;
    int leftovers = 0;

    while (a > 0)
    {
        ans += a;
        int newCandles = (a + leftovers) / b;
        leftovers = (a + leftovers) % b;
        a = newCandles;
    }

    cout << ans << '\n';
    return 0;
}

File: ANewYearCandles.java
import java.util.Scanner;

public class ANewYearCandles {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int a = sc.nextInt();
        int b = sc.nextInt();
        sc.close();
        int ans = 0;
        int leftovers = 0;
        while (a > 0) {
            ans += a;
            int newCandles = (a + leftovers) / b;
            leftovers += (a + leftovers) % b;
            a = newCandles;
        }
        System.out.println(ans);
    }
}